selene_sdk/samplers/random_positions_sampler.py

```diff
@@ -151,86 +159,214 @@ def __init__(self,
             save_datasets=save_datasets,
             output_dir=output_dir)
 
-        self._sample_from_mode = {}
-        self._randcache = {}
-        for mode in self.modes:
-            self._sample_from_mode[mode] = None
-            self._randcache[mode] = {"cache_indices": None, "sample_next": 0}
-
         self.sample_from_intervals = []
         self.interval_lengths = []
 
-        if self._holdout_type == "chromosome":
-            self._partition_genome_by_chromosome()
+        self._num_chroms = 0
+        self._genome_n_bases = 0
+        # Holds the size of the parition of each mode (the portion that is
+        # specified to belong to each mode)
+        self._parititon_sizes = {"validate" : 0, "train" : 0, "test" : 0}
```

Comment:
Shouldn't this be the same as `train_size`, `validation_size` etc? What is the difference?

Reply:
So for ex. if `validate` takes `chm9` and `chm8`, this data structure would hold len(chm9) + len(chm8). This was just done to initialize the np arrays of the original partitions (how we split the 3bln array, after which we take the first `train_size`, etc) So the number we are sampling from each VS. the number of items to sample from, given the partitions themselves. Does that make sense?

Reply:
Can we make this clearer in the variable names and/or the comment line that you have above it?